### ISSUE TITLE
All fields of a Capsule that are of capsule type must have @local or @imports annotations

### ISSUE THREAD

**Issue by TErenberger** [hridesh/panini#68]
If a capsule references another capsule, it must be annotated with @local or @imports. 

```
foreach field on capsule template {
  if field's type elem is capsule interface and not annotated with @local or @imports
    error
  else if there doesn't exists a template currently being compiled whose prefix matches the field's type
    error
  end if
}
```

for this we will need to have a list populated at the start of the annotation processor of all the templates that are being built.

**Comment by dwtj** [comment]
I think this should be rolled into issue #52.

**Comment by dwtj** [comment]
Resolved by bcb6bb949dd2569e1b027c98b88bd9b87ea41506
